ASCIIEngine/Components/RigidBody2D.cs

```diff
@@ -0,0 +1,38 @@
+﻿using ASCIIEngine.Core.BasicClasses;
+
+namespace ASCIIEngine.Core.Components
+{
+    public class RigidBody2D : Component
+    {
+        public Vector2D Direction
+        {
+            get => _direction.Normalize();
+            set => _direction = value;
+        }
+
+        public int Velocity { get; set; }
+ 
+        private Vector2D _direction;
+        private GameObject _parent;
+
+        public RigidBody2D(GameObject parent)
+        {
+            _parent = parent;
+        }
+
+        internal override void Update()
+        {
+            _parent.Position += _direction * Velocity;
+        }
+
+        internal void OnCollision(Vector2D freePosition)
+        {
+            _parent.Position = freePosition;
```

Comment:
In case of a collision the RigidBody must move the object to an unoccupied cell in the direction of the object's last position

Reply:
Object may haven't any last position, so we can't be sure that last position is can help. Here I move objects apart. 
Also we can provide unoccupied cell from Core, where we have access to the map, but i'm not sure that it is should be implemented in this PR.

Reply:
If the object doesn't have a last position, then we move it to any unoccupied cell nearby.
Getting nearby unoccupied cell is the part of collision system, so it should be implemented as a part of this PR.

Reply:
Resolved